src/cromshell/status/command.py

```diff
@@ -114,34 +105,50 @@ def main(config, workflow_id):
     return ret_val
 
 
-def get_metadata_status_summary(workflow_metadata):
-    """Get the status for each call in a workflow
-    and the frequency of those statuses"""
-    # workflow_metadata holds the workflow metadata as a dictionary
-    tmp_execution_status = []
-
-    # For each call in the metadata dictionary create a shortened summary containing
-    # the call name and status
-    for call in workflow_metadata["calls"]:
-        call_element = f'"{call}": '
-        execution_statuses = []
-
-        # For each call name add the number of execution_status to a list
-        for instanceDic in workflow_metadata["calls"][call]:
-            execution_statuses.append(instanceDic["executionStatus"])
-
-        # Create a key and value pair for the execution status and number of times
-        # it is seen
-        status_count = Counter(execution_statuses)
-        # For each status for the call add the call name and status name and count
-        # to list
-        for status in set(execution_statuses):
-            status_element = f'{{"{status}": {status_count[status]}}}'
-            tmp_execution_status.append(f"{call_element}{status_element}")
-
-    tmp_execution_status_json = ", ".join(tmp_execution_status)
-    tmp_execution_status_json = "{" + tmp_execution_status_json + "}"
-    return [json.loads(tmp_execution_status_json)]
+def check_for_failure(metadata: dict):
+    """Checks a workflow metadata dictionary for failing statuses
+    Returns True to indicate workflow or some task(s) has failed"""
+
+    # If the given dictionary contains a 'status' key and has value of "Failed"
+    # then exit the function returning "True" to indicate workflow has failed
+    if metadata.get("status") == "Failed":
+        return True
+
+    # If the dictionary does not contain a failed value for its status key or
+    # if status key does does not exist then we'll iterate through the dictionary
+    # value in search of a nested dictionary (indicating subworkflows)
+    # or a list which holds the task statuses.
+    for value in metadata.values():
+
+        # If a dictionary value is encountered then Depth First Search recursion
+        # is used to traverse the dictionary by iterating through the given dictionary
+        # and reporting whether it found a failure. The reason for this is due to the
+        # tree structure of the metadata dictionary which holds the task statuses
+        # in second layer dictionary key called calls.
```

Comment:
"second layer" is probably too specific.
My concern is that this code was developed with a specific example test data, so any comments that are too specific is going to confuse future developers.

Reply:
replacing it to "a nested"